ethToElrond/bridgeExecutors/signaturesHolder_test.go

```diff
@@ -0,0 +1,272 @@
+
+		sh.clearStoredSignatures()
+
+		assert.Equal(t, 0, len(sh.Signatures([]byte("eth msg"))))
+	})
+	t.Run("same signatures should return the unique ones", func(t *testing.T) {
+		msg := &core.SignedMessage{
+			Payload:        []byte("payload"),
+			Signature:      []byte("sig"),
+			PublicKeyBytes: []byte("pk"),
+			Nonce:          1,
+		}
+		ethMsg := &core.EthereumSignature{
+			Signature:   []byte("eth sig"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		msg2 := &core.SignedMessage{
+			Payload:        []byte("payload2"),
+			Signature:      []byte("sig2"),
+			PublicKeyBytes: []byte("pk2"),
+			Nonce:          1,
+		}
+		ethMsg2 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 2"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		msg3 := &core.SignedMessage{
+			Payload:        []byte("payload3"),
+			Signature:      []byte("sig3"),
+			PublicKeyBytes: []byte("pk3"),
+			Nonce:          1,
+		}
+		ethMsg3 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 2"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		sh := newSignatureHolder()
+		sh.ProcessNewMessage(msg, ethMsg)
+		sh.ProcessNewMessage(msg2, ethMsg2)
+		sh.ProcessNewMessage(msg3, ethMsg3)
+
+		compareBytesSlicesLists(t, [][]byte{[]byte("eth sig"), []byte("eth sig 2")}, sh.Signatures([]byte("eth msg")))
+	})
+	t.Run("same signatures should return filter by message", func(t *testing.T) {
+		msg := &core.SignedMessage{
+			Payload:        []byte("payload"),
+			Signature:      []byte("sig"),
+			PublicKeyBytes: []byte("pk"),
+			Nonce:          1,
+		}
+		ethMsg := &core.EthereumSignature{
+			Signature:   []byte("eth sig"),
+			MessageHash: []byte("eth msg 1"),
+		}
+
+		msg2 := &core.SignedMessage{
+			Payload:        []byte("payload2"),
+			Signature:      []byte("sig2"),
+			PublicKeyBytes: []byte("pk2"),
+			Nonce:          1,
+		}
+		ethMsg2 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 2"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		msg3 := &core.SignedMessage{
+			Payload:        []byte("payload3"),
+			Signature:      []byte("sig3"),
+			PublicKeyBytes: []byte("pk3"),
+			Nonce:          1,
+		}
+		ethMsg3 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 3"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		sh := newSignatureHolder()
+		sh.ProcessNewMessage(msg, ethMsg)
+		sh.ProcessNewMessage(msg2, ethMsg2)
+		sh.ProcessNewMessage(msg3, ethMsg3)
+
+		compareBytesSlicesLists(t, [][]byte{[]byte("eth sig 2"), []byte("eth sig 3")}, sh.Signatures([]byte("eth msg")))
+	})
+}
+
+func compareSignedMessageLists(t *testing.T, list1 []*core.SignedMessage, list2 []*core.SignedMessage) {
+	require.Equal(t, len(list1), len(list2))
+	for _, obj1 := range list1 {
+		found := false
+		for _, obj2 := range list2 {
+			if reflect.DeepEqual(obj1, obj2) {
+				found = true
+			}
+		}
+		require.True(t, found)
+	}
+
+	for _, obj2 := range list2 {
```

Comment:
isn't this useless?
as list1's length is equal to list2's length and all the items in list1 exist in list2, there's no point to check list2 as compared to list1 as well

Reply:
think of this scenario involving 2 slices (does not apply to maps as the keys are unique)
list1: "a", "b", "c", "c" # this can happen due to a bug
list2: "a", "b", "c", "d"

without the second iteration, the test passes.